### ISSUE TITLE
Cannot install v0.6.1 on Minikube

### ISSUE THREAD

**Issue by cdraper-CS** [kubeflow/kubeflow#3812]
/kind bug

**What steps did you take and what happened:**
First I tried to install following the instructions here:
https://www.kubeflow.org/docs/started/getting-started-minikube/
They were clearly out of date and although I finally got 0.5.1 working, I could not install 0.6.1.

Then I tried:
```
minikube start
KUBEFLOW_REPO=$(pwd)  kfctl init localapp --platform minikube
cd localapp
kfctl generate all
kfctl apply all
Error: couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from application Error: StatefulSet.apps "application-controller-stateful-set" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
```
I tried rerunning the `apply` and got the issue noted in #3810. Removing the `istio-ingressgateway` resource yaml and trying again got me back to the original error.

**What did you expect to happen:**

Successful install of kubeflow

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): manifests has 0.6.1-rc.2
- kfctl version: v0.6.1-rc.2-1-g3a37cbc6
- Kubernetes platform: minikube
- Kubernetes version: 1.14
- OS (e.g. from `/etc/os-release`): MacOS 10.14.6

**Comment by AlexZhangHF** [comment]
Hi, Have you ever met issue similiar to #3813 while installing 0.5.1? Thx.

**Comment by jlewi** [comment]
If you want to install on minikube instead of using platform I would suggest starting minikube and then following the instructions for installing on an existing cluster.

Here are updated instructions (sorry the PR hasn't merged yet)
https://deploy-preview-977--competent-brattain-de2d6d.netlify.com/docs/k8s/

**Comment by jlewi** [comment]
@kkasravi  the error

```
Error: couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from application Error: StatefulSet.apps "application-controller-stateful-set" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string(nil): `selector` does not match template `labels`]
```

Looks like a problem in the kustomize manifest for the application controller.   Is there a bug in the manifest? 

https://github.com/kubeflow/manifests/blob/master/application/application/base/stateful-set.yaml

It doesn't look like we specify any labels or a selector.

It does look like we apply some commonLabels in one of the overlays
https://github.com/kubeflow/manifests/blob/master/application/application/overlays/application/kustomization.yaml

It looks like GCP config is using that overlay
https://github.com/kubeflow/kubeflow/blob/b229c1beef4648b1ab29b252c90fe318fd84bc7c/bootstrap/config/kfctl_gcp_iap.yaml#L51

So I suspect that is the problem.

**Comment by xaniasd** [comment]
not fully onboard with the code structure yet so please any blatant ignorance: don't you need a selector (or labels) for a StatefulSet? [spec here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#statefulsetspec-v1beta1-apps)

**Comment by jlewi** [comment]
@xaniasd yes the selector is currently applied via an overlay and not part of the base kustomize package. So I think the fix is to add it to the base here
https://github.com/kubeflow/manifests/blob/master/application/application/base/stateful-set.yaml

@xaniasd Want to submit a PR to fix it?

**Comment by xaniasd** [comment]
sure thing!

**Comment by nkumar15** [comment]
@xaniasd, not sure where I am doing the mistake, but I am still facing the same issue. Any suggestions pls?

`couldn't apply KfApp:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: couldn't create resources from application Error: no matches for kind "Application" in version "app.k8s.io/v1beta1"
`
minikube version: v1.1.0
kfctl v0.6.1-rc.2-1-g3a37cbc6
kubectl version 1.14

**Comment by xaniasd** [comment]
@nkumar15, it seems to be a different error message.

**Comment by nkumar15** [comment]
@xaniasd , should I open a separate issue for it? I see #3954 is closed citing #3812 as duplicate.
